pkg/controller/machine/controller.go

```diff
@@ -99,38 +101,33 @@ func (c *MachineControllerImpl) Reconcile(machine *clusterv1.Machine) error {
 			return nil
 		}
 		glog.Infof("reconciling machine object %v triggers delete.", name)
-		if err := c.delete(machine); err != nil {
+		if err := c.delete(m); err != nil {
 			glog.Errorf("Error deleting machine object %v; %v", name, err)
 			return err
 		}
 
 		// Remove finalizer on successful deletion.
-		glog.Infof("machine object %v deletion successful, removing finalizer.", name)
-		machine.ObjectMeta.Finalizers = util.Filter(machine.ObjectMeta.Finalizers, clusterv1.MachineFinalizer)
-		if _, err := c.clientSet.ClusterV1alpha1().Machines(machine.Namespace).Update(machine); err != nil {
-			glog.Errorf("Error removing finalizer from machine object %v; %v", name, err)
-			return err
-		}
```

Comment:
Can you add the finalizer removal in the existing machine actuators? Otherwise, machines will fail to delete after this PR merges, right?

Reply:
i checked the gce and vsphere machine actuator and both already removes the finalizer